src/components/Table/components/EditColumnsButton.tsx

```diff
@@ -76,22 +74,27 @@ export function EditColumnsButton<R extends Kinded>(props: EditColumnsButtonProp
     <OverlayTrigger {...props} menuTriggerProps={menuTriggerProps} state={state} buttonRef={buttonRef} {...tid}>
       <div
         css={{
-          ...Css.bgWhite.py5.px3.maxwPx(380).bshBasic.$,
+          ...Css.dg.gtc("1fr auto").gap2.bgWhite.p2.maxwPx(326).$,
           "&:hover": Css.bshHover.$,
         }}
       >
-        <div css={Css.gray500.xsSb.mb1.ttu.$}>{title || "Select columns to show"}</div>
-        <CheckboxGroup
-          label={title || "Select columns to show"}
-          onChange={(values) => setSelectedValues(values)}
-          values={selectedValues}
-          options={options}
-          columns={2}
-          labelStyle="hidden"
-        />
-        <div css={Css.mt1.$}>
-          <Button variant={"tertiary"} label={"Clear selections"} onClick={() => setSelectedValues([])} />
-        </div>
+        {options.map((option) => (
+          <Fragment key={option.value}>
+            <div css={Css.sm.truncate.pr1.$}>{option.label}</div>
+            <Switch
+              compact
+              selected={selectedValues.includes(option.value)}
+              onChange={(value) =>
+                setSelectedValues(
+                  value ? [...selectedValues, option.value] : selectedValues.filter((v) => v !== option.value),
+                )
+              }
+              labelStyle="hidden"
+              label={""}
+              {...tid[`option${option.value}`]}
```

Comment:
`data-testids` are fine/meant to be static, i.e not dynamic, so just `tid.option` would be fine, and tests can use "get the Nth option" instead of "get the data-test-id with value xyz".

Reply:
...I guess we do end up with `data-testid=firstName` and `data-testid=lastName` for things like textboxes, it's not like every textbox is `data-testid=input`.

Yeah, I guess nevermind, this lgtm